### DISCUSSION TITLE
Some issues with latest dependencies

### DISCUSSION THREAD

**Discussion by shanoasaber** [Nebukam/com.nebukam.orca#11] [Q&A]
I modified dependencies in package.json to use the latest libs of DOTS. no error occured. But have some problem. The agent often stun and Samples not work correctly. They can't move to another side when they touch the wall line.
Have you try out what the differences between new and old burst and math libs ?

Unity Version: 2022.3.17f1c1
com.unity.collections: 2.4.1
com.unity.mathematics: 1.3.1
com.unity.burst: 1.8.15

orca packages.json:
![image](https://github.com/user-attachments/assets/286eafb9-1235-4393-826e-7fc8e86ab9ab)

job-assist packages.json:
![image](https://github.com/user-attachments/assets/0bf2b719-f6c4-42ca-a162-e2f9249166ab)

common packages.json:
![image](https://github.com/user-attachments/assets/96bcfa33-21a4-44a6-8d2a-77947fe6656a)

**Comment by Nebukam** [comment]
Hello @shanoasaber!
I tested on my side and updated the packages so they use the latest burst (`1.8.17`) and mathematics (`1.3.1`) and did not encounter any issue.
I tested with entities & collections `2.1.4` and did not get any problem, and there is no reason it would break with `2.4.1` -- something wrong is going on with your project and I'm not sure what.

I suspect the agents aren't being updated properly and the update order of agents is not maintained: DOTS shuffles memory around so you can't update agents 1:1 between updates depending on your implementation, you need to maintain your own map of agents/entites.

**Reply by shanoasaber** [reply]
Thank you for your reply. I'll retest it in a blank project. I'll write new reply if issues always exists and send you project or some video.

**Comment by Nebukam** [comment]
Side note: there is no need to update the com.unity.xxx package versions, the editor will use the one from your project even if they mismatch :)

**Comment by shanoasaber** [comment]
https://github.com/user-attachments/assets/c8ba3ce3-ff32-4283-94c2-0ea925453142

This video show how the agent was stuck aside a wall. and this is sample scene of this project i've never modified.
This issue also occured in my project when an agent was surrounded by several other agents. even if other agents gone away. the agent still stuck at same place.

**Reply by Nebukam** [reply]
Hmmm indeed this is odd -- I think I know why this is happening I'll look into it.

I believe the problem comes from open, concave shapes: I did an update a while ago and forgot to check the other examples. 
Can you confirm the main example works properly on your setup? The one with a lot of agents inside a square and a bunch of random obstacles

**Comment by Nebukam** [comment]
Hey there @shanoasaber !
Quick update on this: turns out I changed some code a while ago to force all obstacles to be closed but in the poorest way possible.
The debug display is currently wrong, this is what the actual, under-the-hood collision looks like, which explains the behavior you're seeing.

![image](https://github.com/user-attachments/assets/27fb1023-1bfe-410d-9d6e-2e7998435f76)


I'll be working on fixing this ASAP, will keep you posted here.
Thanks a lot for pointing this out!

**Reply by shanoasaber** [reply]
Thank you for your continued updates. 
I've been so busy with the project going live lately I've been forgetting about it, but hopefully my thanks aren't too late!
This library is brilliantly written and very helpful. I think positive feedback on issues will make it even better :)

**Comment by shanoasaber** [comment]
I'm waiting for an updates. Did i miss something ?

**Comment by Nebukam** [comment]
You did not, I hadn’t had the time to look into it just yet!

Le mer. 2 oct. 2024 à 21:35, shanoasaber ***@***.***> a
écrit :
